src/main/java/com/kalessil/phpStorm/phpInspectionsEA/inspectors/codeSmell/DisallowWritingIntoStaticPropertiesInspector.java

```diff
@@ -0,0 +1,107 @@
+package com.kalessil.phpStorm.phpInspectionsEA.inspectors.codeSmell;
+
+import com.intellij.codeInspection.ProblemHighlightType;
+import com.intellij.codeInspection.ProblemsHolder;
+import com.intellij.openapi.project.Project;
+import com.intellij.psi.PsiElementVisitor;
+import com.intellij.util.containers.MultiMap;
+import com.jetbrains.php.PhpIndex;
+import com.jetbrains.php.lang.psi.elements.*;
+import com.kalessil.phpStorm.phpInspectionsEA.openApi.BasePhpElementVisitor;
+import com.kalessil.phpStorm.phpInspectionsEA.openApi.BasePhpInspection;
+import com.kalessil.phpStorm.phpInspectionsEA.utils.ExpressionSemanticUtil;
+import com.kalessil.phpStorm.phpInspectionsEA.utils.PhpIndexUtil;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Collection;
+
+
+public class DisallowWritingIntoStaticPropertiesInspector extends BasePhpInspection {
+
+    private static final String message = "Static property should be modified only inside the source class";
+
+    @NotNull
+    public String getShortName() {
+        return "DisallowWritingIntoStaticPropertiesInspection";
+    }
+
+    @NotNull
+    @Override
+    public PsiElementVisitor buildVisitor(@NotNull ProblemsHolder holder, boolean b) {
+
+
+        return new BasePhpElementVisitor() {
+
+            @Override
+            public void visitPhpAssignmentExpression(AssignmentExpression assignmentExpression) {
+                PhpPsiElement variable = assignmentExpression.getVariable();
+                if (!(variable instanceof FieldReference)) {
+                    return;
+                }
+
+                if (!((FieldReference) variable).getReferenceType().isStatic()) {
```

Comment:
And not a constant, this would be invalid but we still don't want to process constant references.